packages/hooks/src/useAnalytics.ts

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { Analytics, TEvents } from '@deriv/analytics';
+import { useStore } from '@deriv/stores';
+import { removeNullOrUndefinedValues } from '@deriv/shared';
+
+export type TTrackRealAccountSignup = Omit<
+    TEvents['ce_real_account_signup_form'],
+    'landing_company' | 'user_choice'
+> & {
+    user_choice?: string | object;
+};
+
+const useAnalytics = () => {
+    const { ui, client } = useStore();
+    const { is_mobile, real_account_signup_target } = ui;
+
+    React.useEffect(() => {
+        client?.loginid &&
+            Analytics.setAttributes({
+                device_type: is_mobile ? 'mobile' : 'desktop',
+                account_type: client?.loginid?.slice(0, 2),
+            });
+    }, [is_mobile, client?.loginid]);
+
+    const trackRealAccountSignup = React.useCallback(
+        ({
+            action,
+            step_codename,
+            step_num,
+            user_choice,
+            real_signup_error_message,
+            form_source,
+        }: TTrackRealAccountSignup) => {
+            if (real_account_signup_target === 'maltainvest') return;
+
+            const payload: TEvents['ce_real_account_signup_form'] = {
+                action,
+                step_codename,
+                step_num,
+                user_choice: JSON.stringify(user_choice),
+                form_source: form_source ?? window?.location?.href,
+                form_name: 'real_account_signup_form',
+                real_signup_error_message: JSON.stringify(real_signup_error_message),
+                landing_company: real_account_signup_target,
+            };
+            Analytics?.trackEvent('ce_real_account_signup_form', removeNullOrUndefinedValues(payload));
```

Comment:
in object destructure like approach we doesn't need this func

Reply:
As far as I've seen, event with destructuring, if a value is not specified, that value will be undefined with the key specified in the new object, also we're destructuring from the parameter
<img width="509" alt="Screenshot 2023-11-24 at 11 11 14 AM" src="https://github.com/binary-com/deriv-app/assets/129842155/efc2ba71-5302-448f-afad-6a7a47177e7b">

Reply:
I added this function to remove undefined or null values so we don't send any empty value as it may cause issues with data processing